nuget/ValidationContext.cs

```diff
@@ -0,0 +1,243 @@
+        ///     Removes extracted glTF-Validator temporary executable.
+        /// </summary>
+        public void Dispose()
+        {
+            if (_embeddedValidatorPath == null) return;
+            try
+            {
+                File.Delete(_embeddedValidatorPath);
+            }
+            catch
+            {
+                // ignored
+            }
+        }
+
+        /// <summary>
+        ///     Deserializes validation report from JSON string.
+        /// </summary>
+        /// <param name="json">String containing the report.</param>
+        public static ValidationReport Parse(string json)
+        {
+            return JsonSerializer.Deserialize<ValidationReport>(json, Options);
+        }
+
+        /// <summary>
+        ///     Asynchronously deserializes validation report from JSON stream.
+        /// </summary>
+        /// <param name="json">Stream containing JSON report.</param>
+        /// <param name="cancellationToken">A token that may be used to cancel the operation.</param>
+        public static ValueTask<ValidationReport> ParseAsync(Stream json, CancellationToken cancellationToken = default)
+        {
+            return JsonSerializer.DeserializeAsync<ValidationReport>(json, Options, cancellationToken);
+        }
+
+        /// <summary>
+        ///     Runs glTF-Validator on provided asset and returns deserialized report.
+        /// </summary>
+        /// <param name="path">Path to the glTF file.</param>
+        /// <param name="customValidatorPath">
+        ///     Path to the custom glTF-Validator executable. Must be provided if
+        ///     <c>ValidationContext</c> was created with <c>useEmbeddedValidator = false</c>.
+        /// </param>
+        /// <param name="validateResources">When set to <c>false</c>, only JSON description is validated.</param>
+        /// <param name="timeout">Overrides default validation timeout.</param>
+        /// <returns>A <c>ValidationReport</c> instance.</returns>
+        /// <exception cref="ArgumentException">Thrown when either asset or custom glTF-Validator path is incorrect.</exception>
+        /// <exception cref="FileNotFoundException">Thrown when glTF asset or custom glTF-Validator is not found.</exception>
+        /// <exception cref="InvalidOperationException">
+        ///     Thrown when the <c>ValidationContext</c> was created without embedded
+        ///     validator but custom one was not provided.
+        /// </exception>
+        /// <exception cref="TimeoutException">Thrown when validation takes longer than allowed time.</exception>
+        /// <exception cref="ApplicationException">Thrown when the validator encounters unhandled runtime error.</exception>
+        public ValidationReport Validate(string path, string customValidatorPath = "", bool validateResources = true,
+            TimeSpan timeout = default)
+        {
+            return ValidateAsync(path, customValidatorPath, validateResources, timeout).Result;
```

Comment:
Does this work? I think you have to wait for the task to complete.

Reply:
AFAIU, calling `.Result` on `ValueTask` should handle that internally. Maybe need to double-check.

Reply:
In the Remarks of the [documentation](https://docs.microsoft.com/en-us/dotnet/api/system.threading.tasks.valuetask-1?view=netcore-3.1#remarks), it states:

> The following operations should never be performed on a ValueTask<TResult> instance:
> ...
> Using .Result ... when the operation hasn't yet completed...
> ...

Reply:
Also
> consumers may not read Result until the instance has completed

Reply:
Oops, it's `Task.Result` that waits until completion, not `ValueTask.Result`.
> Accessing the property's get accessor blocks the calling thread until the asynchronous operation is complete; it is equivalent to calling the `Wait` method.